# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/fetch/FetchChild.h
# Commit: 3eb8fdd0ba6d
# Full Hash: 3eb8fdd0ba6d2d921abe85086539b15d1e855686
# Author: Eden Chuang <echuang@mozilla.com>
# Date: 2023-01-12 21:30:33
# Regressor Bug: 1351231
# File Overlap Count: 2
# Description:
#   Bug 1351231 - PFetch protocol declaration and implementation. r=dom-worker-reviewers,jesup
#   
#   Depends on D138813
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D142436
# ==============================================================================

diff -r a1fcf22a2a0e -r 3eb8fdd0ba6d dom/fetch/FetchChild.h
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/dom/fetch/FetchChild.h	Thu Jan 12 17:24:28 2023 +0000
@@ -0,0 +1,40 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef mozilla_dom_fetchChild_h__
+#define mozilla_dom_fetchChild_h__
+
+#include "mozilla/dom/PFetchChild.h"
+
+namespace mozilla::dom {
+
+class FetchChild final : public PFetchChild {
+  friend class PFetchChild;
+
+ public:
+  NS_INLINE_DECL_THREADSAFE_REFCOUNTING(FetchChild, override);
+
+  mozilla::ipc::IPCResult Recv__delete__(const nsresult&& aResult);
+
+  mozilla::ipc::IPCResult RecvOnResponseAvailableInternal(
+      ParentToChildInternalResponse&& aResponse);
+
+  mozilla::ipc::IPCResult RecvOnResponseEnd(ResponseEndArgs&& aArgs);
+
+  mozilla::ipc::IPCResult RecvOnDataAvailable();
+
+  mozilla::ipc::IPCResult RecvOnFlushConsoleReport(
+      nsTArray<net::ConsoleReportCollected>&& aReports);
+
+  mozilla::ipc::IPCResult RecvOnCSPViolationEvent(const nsAString& aJSon);
+
+ private:
+  ~FetchChild() = default;
+
+  void ActorDestroy(ActorDestroyReason aReason) override;
+};
+
+}  // namespace mozilla::dom
+
+#endif